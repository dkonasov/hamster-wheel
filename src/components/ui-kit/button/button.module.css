.variant-round {
    --button-border-radius: 50%;
    --button-background: var(--color-primary);
    --button-text-color: var(--color-text-accent);
    --button-padding: 0;
}

.variant-primary {
    --button-border-radius: 10px;
    --button-background: var(--color-primary);
    --button-text-color: var(--color-text-accent);
    --button-padding: 10px;
}

.variant-secondary {
    --button-border-radius: 10px;
    --button-background: transparent;
    --button-text-color: var(--color-primary);
    --button-padding: 10px;
}

.root {
    border-radius: var(--button-border-radius);
    background: var(--button-background);
    color: var(--button-text-color);
    display: flex;
    padding: var(--button-padding);
    border: none;
}

.root:hover {
    cursor: pointer;
    opacity: .8;
}

.root:active {
    opacity: .7;
}