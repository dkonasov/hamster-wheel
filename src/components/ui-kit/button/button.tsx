import React, { ButtonHTMLAttributes, forwardRef, HTMLAttributes } from "react";
import { FC } from "react";
import cx from "classnames";
import styles from "./button.module.css";

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: "round" | "clear" | "primary" | "secondary";
  type?: "submit" | "button";
}

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => {
    const { className, variant = "primary", ...restProps } = props;
    return (
      <button
        className={cx(className, styles.root, styles[`variant-${variant}`])}
        {...restProps}
        ref={ref}
      >
        {props.children}
      </button>
    );
  }
);
