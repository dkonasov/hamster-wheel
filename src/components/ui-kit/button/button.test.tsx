import React from "react";
import client, { Root } from "react-dom/client";
import { act } from "react-dom/test-utils";
import { Button } from "./button";

let container: HTMLDivElement | null = null;
let root: Root | null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
  root = client.createRoot(container);
});

afterEach(() => {
  act(() => {
    root.unmount();
  });
  root = null;
  container.remove();
  container = null;
});

describe("Button component", () => {
  it("should render button", () => {
    act(() => {
      root.render(<Button>Test</Button>);
    });

    expect(container.innerHTML).toMatchSnapshot();
  });
});
