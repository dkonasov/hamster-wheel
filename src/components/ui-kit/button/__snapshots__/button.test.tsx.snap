// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render button 1`] = `"<button class="button-module__root___1pVgv button-module__variant-primary___3dh1m">Test</button>"`;
